packages/indexer-agent/src/network.ts

```diff
@@ -206,6 +214,25 @@ export class Network {
     return output!
   }
 
+  async getRevertReason(
+    logger: Logger,
+    txRequest: providers.TransactionRequest,
+  ): Promise<string> {
+    let revertReason = 'unknown'
+    try {
+      const code = await this.ethereum.call(txRequest)
+      revertReason = utils.toUtf8String(`0x${code.substr(138)}`)
+    } catch (e) {
+      if (e.body.includes('out of gas')) {
+        revertReason = 'out of gas'
+      } else {
+        throw indexerError(IndexerErrorCode.IE051)
+      }
+    }
+    logger.warn('Transaction reverted:', { reason: revertReason })
+    return revertReason
+  }
+
```

Comment:
Looking at this, I wonder if this is sufficient or if we need to be smarter about detecting the revert reason. See https://github.com/graphprotocol/graph-node/blob/6d9b1ecf1b7b40e16b07c3d465345fea5272a905/chain/ethereum/src/ethereum_adapter.rs#L468-L569 for comparison.

There's also an `eth-revert-reason` that may be worth checking out, rather than rolling our own. https://github.com/authereum/eth-revert-reason

Reply:
I tried `eth-revert-reason` and it was getting errors, so I used it as inspiration to implement in our system.

Reply:
Ah, ok. Let's hope this covers enough cases / Ethereum implementations then. :)

Reply:
I've tested with Alchemy. Maybe I should also test with Infura before merging?